server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -879,7 +884,9 @@ private void executePipelines(
         ingestDocument.resetReroute();
         final String originalIndex = indexRequest.indices()[0];
         final Consumer<Exception> exceptionHandler = (Exception e) -> {
+            FailureStoreMetrics.incrementForIndex(failureStoreMetrics.failureCounter(), originalIndex);
```

Comment:
how about
`failureStoreMetrics.incrementFailureOn(originalIndex)` / `failureStoreMetrics.incrementRedirectOn(originalIndex) `

Reply:
Out of curiosity, what makes you prefer your suggestion over my current implementation? I like that I only have to have one method, but my method invocation line is a bit longer in my version.

Reply:
it looks like you have added a static utility method on a class to operate only on its instances. It would feel more natural to have this as an instance method.

>  I like that I only have to have one method, but my method invocation line is a bit longer in my version.

you have one method, but you can only use it with 2 parameters (failureCounter/redirectionCounter). Any other instance of counter would not be valid. I think it is better to make it explicit with 2 separate method names.

Reply:
Ah okay that makes sense, thanks for the explanation. Will update :+1: